numba/targets/arrayobj.py

```diff
@@ -1838,6 +1838,29 @@ def np_unique_impl(a):
     return np_unique_impl
 
 
+@overload(np.repeat)
+def np_repeat(a):
```

Comment:
The signature for both the typing part (this function) and implementation (the inner function) should be the same, so this should read:
```python
@overload(np.repeat)
def np_repeat(a, repeats, axis=None):
```
Also, for simplicity, I'd recommend leaving out the `axis` kwarg for now until the function is working correctly. Handling kwargs and `None` can be challenging and so probably should be left until the end.